src/api/projects.js

```diff
@@ -67,7 +67,95 @@ export const create = (event, context, callback) => {
     }).catch((error) => {
       return callback(null, done({
         statusCode: 500,
-        message: error.message,
+        message: error,
       }));
     });
 };
+
+/**
+ * Add user to a given project
+ * @param {*} event
+ * @param {*} context
+ * @param {*} callback
+ * @return {function} callback
+ */
+export const addUser = (event, context, callback) => {
+  const userId = JSON.parse(event.body).userId;
+  const projectId = JSON.parse(event.body).projectId;
+
+  if (
+    userId == null || userId.trim() == '' ||
+    projectId == null || isNaN(projectId)
+  ) {
+    return callback(null, done({
+      statusCode: 400,
+      message: `Request doesn't contain a valid object`,
+    }));
+  }
+
+  const uuid = event.requestContext.authorizer.principalId;
+  Permission.hasPermission(uuid, {realm: 'project', action: 'insert'})
+    .then((confirmation) => {
+      if (!confirmation) {
+        return callback(null, done({
+          statusCode: 403,
+          message: `User doesn't have enough permission to perform this action`,
+        }));
+      };
+
+       User.find({
+        attributes: ['uuid', 'organizationId'],
+        where: {
+          uuid: userId,
+        },
+      }).then((user) => {
+        Project.find({
+          where: {
+            id: projectId,
+          },
+        }).then((project) => {
+            if (project.organizationId == user.organizationId) {
+              project.addUser(user).then((result) => {
+                if (result.length == 0) {
+                  return callback(null, done({
+                    statusCode: 400,
+                    message: `User has already added to this project`,
+                  }));
+                } else {
+                  return callback(null, done(null, {
+                    statusCode: 200,
+                    data: result[0][0],
+                  }));
+                }
+              }).catch((error) => {
+                return callback(null, done({
+                  statusCode: 500,
+                  message: error,
+                }));
+              });
+            } else {
+              return callback(null, done({
+                statusCode: 400,
+                message: `User and project not belong to the same organization`,
```

Comment:
Don't do this, people can do request with random ids and get project details. Just say bad request or project not found. Fixing the above mentioned where clause issue will resolve this one as well